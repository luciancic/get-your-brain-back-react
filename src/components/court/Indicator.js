import React from 'react';

const Indicator = ({ children, displaySide}) => {
    return <span className={ displaySide } style={{ padding: '0 15px'}}>{ children }</span>
}

export default Indicator;