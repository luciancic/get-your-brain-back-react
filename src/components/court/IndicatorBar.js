import React from 'react';
import Indicator from './Indicator';

function renderRemainingRounds(maxRounds, remainingRounds) {
    return remainingRounds > maxRounds ? 
        <span>{maxRounds}<span className="green-text"> + {remainingRounds - maxRounds}</span></span>
        : remainingRounds
}

export default ({ maxRounds, remainingRounds, n }) => (
    <div>
        <Indicator displaySide="left">Remaining rounds: { renderRemainingRounds(maxRounds, remainingRounds) }</Indicator>
        <Indicator displaySide="right">N: {n}</Indicator>
    </div>
)
