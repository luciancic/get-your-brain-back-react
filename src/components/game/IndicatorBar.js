import React from 'react';
import Indicator from './Indicator';
import RemainingRoundsIndicator from './RemainingRoundsIndicator';

export default ({ maxRounds, remainingRounds, n }) => (
    <div style={{ margin: '15px 0' }}>
        <RemainingRoundsIndicator maxRounds={maxRounds} remainingRounds={remainingRounds}/>
        <Indicator displaySide="right">N: {n}</Indicator>
        <br/>
    </div>
)
