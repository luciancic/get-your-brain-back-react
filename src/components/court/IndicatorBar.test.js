import React from 'react';
import { shallow, mount } from 'enzyme';
import IndicatorBar from './IndicatorBar';
import Indicator from './Indicator';

it('renders IndicatorBar', () => {
  const wrapper = shallow( <IndicatorBar maxRounds={20} remainingRounds={22} n={2} /> )
  expect( wrapper ).toMatchSnapshot();
  expect( wrapper.find('span') ).toHaveLength(2);
  
  wrapper.setProps( { remainingRounds: 20 } );
  expect( wrapper.find('span') ).toHaveLength(0);
});