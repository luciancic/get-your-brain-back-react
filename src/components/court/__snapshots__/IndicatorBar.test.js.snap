// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IndicatorBar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    maxRounds={20}
    n={2}
    remainingRounds={22}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Indicator
          displaySide="left"
        >
          Remaining rounds: 
          <span>
            20
            <span
              className="green-text"
            >
               + 
              2
            </span>
          </span>
        </Indicator>,
        <Indicator
          displaySide="right"
        >
          N: 
          2
        </Indicator>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Remaining rounds: ",
            <span>
              20
              <span
                className="green-text"
              >
                 + 
                2
              </span>
            </span>,
          ],
          "displaySide": "left",
        },
        "ref": null,
        "rendered": Array [
          "Remaining rounds: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                20,
                <span
                  className="green-text"
                >
                   + 
                  2
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              20,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " + ",
                    2,
                  ],
                  "className": "green-text",
                },
                "ref": null,
                "rendered": Array [
                  " + ",
                  2,
                ],
                "type": "span",
              },
            ],
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "N: ",
            2,
          ],
          "displaySide": "right",
        },
        "ref": null,
        "rendered": Array [
          "N: ",
          2,
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Indicator
            displaySide="left"
          >
            Remaining rounds: 
            <span>
              20
              <span
                className="green-text"
              >
                 + 
                2
              </span>
            </span>
          </Indicator>,
          <Indicator
            displaySide="right"
          >
            N: 
            2
          </Indicator>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Remaining rounds: ",
              <span>
                20
                <span
                  className="green-text"
                >
                   + 
                  2
                </span>
              </span>,
            ],
            "displaySide": "left",
          },
          "ref": null,
          "rendered": Array [
            "Remaining rounds: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  20,
                  <span
                    className="green-text"
                  >
                     + 
                    2
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                20,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " + ",
                      2,
                    ],
                    "className": "green-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    " + ",
                    2,
                  ],
                  "type": "span",
                },
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "N: ",
              2,
            ],
            "displaySide": "right",
          },
          "ref": null,
          "rendered": Array [
            "N: ",
            2,
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
